src/gauging.jl

```diff
@@ -27,13 +26,17 @@ function vidal_gauge(
     vsrc, vdst = src(e), dst(e)
     ψvsrc, ψvdst = ψ_vidal[vsrc], ψ_vidal[vdst]
 
-    pe = partitionedge(pψψ, (vsrc, 1) => (vdst, 1))
+    pe = partitionedge(partitioned_itensornetwork(bpc), (vsrc, 1) => (vdst, 1))
```

Comment:
This is a bit unfortunate. Can we make `BeliefPropagationCache` an `AbstractPartitionedGraph` and make these functions work automatically? Or at least forward these kinds of definitions?

Reply:
My only concern with making a `BeliefPropagationCache` and `AbstractPartitionedGraph` was that if, down the line we introduced an `AbstractCache` type and wanted to unify the different cache types under that umbrella

Reply:
At the moment I am not sure what the `AbstractCache` type would look like so maybe that is not a worry?

Reply:
That's a good point. We may not actually need an `AbstractCache` type, and just allow any object to be a cache if you define the right overloads for it. But for now, let's at least define `partitionedge` for `BeliefPropagationCache` by forwarding to `partitioned_itensornetwork` to make this code simpler, and think about the type hierarchy in future PRs.

Related to that, my guess is if we just loosen type constraints in many of the partitioned graph functions to allow them to accept anything as inputs, then a lot of partitioned graph functionality could "just work" as long as some minimal number of functions were defined, even on a non-graph type like the current `BeliefPropagationCache`. So we could think of many of the partitioned graph functionality as more of an interface rather than written specifically for the `PartitionedGraph` type or `AbstractPartitionedGraph` subtypes.

Specifically, say we took all of this `PartitionedGraph` code: https://github.com/mtfishman/NamedGraphs.jl/blob/v0.1.21/src/Graphs/partitionedgraphs/partitionedgraph.jl#L46-L95 and the `AbstractPartitionedGraph` code: https://github.com/mtfishman/NamedGraphs.jl/blob/v0.1.21/src/Graphs/partitionedgraphs/abstractpartitionedgraph.jl (besides the definitions for pure `Graphs.jl`/`AbstractGraph` functionality) and removed the `::PartitionedGraph` and `::AbstractPartitionedGraph` type constraints. Then they could "just work" for `BeliefPropagationCache` without making it an `AbstractPartitionedGraph` subtype, as long as it overloads some minimal interface which makes it a partitioned graph.

Reply:
Okay for now I will just forward the relevant functions but I agree with you that it seems a lot of the `PartitionedGraph` code is not really specific to the type and more interface-like. In some future PR we could try loosening things that way and see what happens.